driver/src/main.rs

```diff
@@ -73,6 +73,11 @@ struct Options {
     #[structopt(long, env = "SOLVER_TYPE", default_value = "naive-solver")]
     solver_type: SolverType,
 
+    /// Which optimization solver the solver should use internally. Is passed as
+    /// `--solver` to the solver.
+    #[structopt(long, env = "SOLVER_INTERNAL_SOLVER")]
+    solver_internal_solver: Option<String>,
```

Comment:
Would it make sense to validate that if the option is set, it's a valid choice (e.g. either gurobi or scip)?
I don't think our selection will change anytime soon and it might lead to weird unexpected behavior if there is a typo in the string (e.g. we think we use gurboi but fall back to scip or error when trying to solve the batch after one day of inactivity). Having an explicit type check at parsing time could be valuable imo.

Reply:
I was debating this:

> I intentionally did not create an enum for the internal solver type
because then we would end up duplicating this from the solver. Instead
we just forward the argument.

I have no strong opinion either way. I'll see if someone else has an opinion on this and if not I'll add the enum.

Reply:
I think since we already validate other parameters, we should probably do it here as well. Also, I think that since we are handling a specific solver argument in this case we should probably limit its scope to known good values. It would also enable us to make decisions in the driver (if needed) based on the optimizer implementation.

I do see your point about the fact that is just passed to the solver directly. In that case maybe we can have a `solver_args` option that just forwards all specified parameters as is to the solver (kind of like a `-Z` flag for C/C++ compilers for specifying linker arguments). This might be useful if we need some "emergency" arguments on one of our deployments that the driver doesn't have support for.